sharding-jdbc/sharding-jdbc-core/src/test/java/org/apache/shardingsphere/shardingjdbc/jdbc/core/statement/ShardingSpherePreparedStatementTest.java

```diff
@@ -316,6 +320,77 @@ public void assertAddBatchWithGenerateKeyColumn() throws SQLException {
         }
     }
     
+    @Test
+    public void assertAddOnDuplicateKey() throws SQLException {
+        int itemId = 1;
+        int userId1 = 101;
+        int userId2 = 102;
+        int orderId = 200;
+        String status = "init";
+        String updatedStatus = "updated on duplicate key";
+        try (Connection connection = getShardingSphereDataSource().getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(INSERT_ON_DUPLICATE_KEY_SQL);
+             PreparedStatement queryStatement = connection.prepareStatement(SELECT_SQL_WITH_PARAMETER_MARKER_RETURN_STATUS)) {
+            preparedStatement.setInt(1, itemId);
+            preparedStatement.setInt(2, orderId);
+            preparedStatement.setInt(3, userId1);
+            preparedStatement.setString(4, status);
+            preparedStatement.setInt(5, itemId);
+            preparedStatement.setInt(6, orderId);
+            preparedStatement.setInt(7, userId2);
+            preparedStatement.setString(8, status);
+            preparedStatement.setString(9, updatedStatus);
+            int result = preparedStatement.executeUpdate();
+            assertThat(result, is(2));
+            
```

Comment:
Please remove useless blank lines.
If we think use blank line to separate diff logic block, it is better extract them as a new private method.